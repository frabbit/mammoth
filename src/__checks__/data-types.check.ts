import {
  arrayAgg,
  bigint,
  bigserial,
  bit,
  bitVarying,
  bool,
  boolean,
  box,
  bytea,
  caseInsensitiveText,
  char,
  character,
  characterVarying,
  cidr,
  circle,
  citext,
  coalesce,
  count,
  date,
  decimal,
  defineDb,
  defineTable,
  doublePrecision,
  enumType,
  float4,
  float8,
  inet,
  int,
  int2,
  int4,
  int8,
  integer,
  interval,
  json,
  jsonb,
  line,
  lseg,
  macaddr,
  macaddr8,
  money,
  numeric,
  path,
  pgLsn,
  point,
  polygon,
  real,
  serial,
  serial2,
  serial4,
  serial8,
  smallint,
  smallserial,
  sum,
  text,
  time,
  timeWithTimeZone,
  timeWithoutTimeZone,
  timestamp,
  timestampWithTimeZone,
  timestampWithoutTimeZone,
  timestamptz,
  tsquery,
  tsvector,
  txidSnapshot,
  uuid,
  varbit,
  varchar,
  xml,
} from '../../.build';

import { Query } from '../../.build/query';
import { ResultSet } from '../../.build/result-set';

const toSnap = <T extends Query<any>>(query: T): ResultSet<T, true> => {
  return undefined as any;
};

/** @dts-jest enable:test-type */

const foo = defineTable({
  id: uuid().primaryKey().default(`gen_random_uuid()`),
  int8: int8(),
  bigint: bigint(),
  bigserial: bigserial(),
  serial8: serial8(),
  bit: bit(),
  bitVarying: bitVarying(),
  varbit: varbit(),
  boolean: boolean(),
  bool: bool(),
  box: box(),
  bytea: bytea(),
  character: character(),
  char: char(),
  characterVarying: characterVarying(),
  varchar: varchar(),
  circle: circle(),
  cidr: cidr(),
  caseInsensitiveText: caseInsensitiveText(),
  citext: citext(),
  date: date(),
  doublePrecision: doublePrecision(),
  float8: float8(),
  inet: inet(),
  decimal: decimal(),
  integer: integer(),
  int: int(),
  int4: int4(),
  interval: interval(),
  json: json(),
  jsonb: jsonb(),
  line: line(),
  lseg: lseg(),
  macaddr: macaddr(),
  macaddr8: macaddr8(),
  money: money(),
  numeric: numeric(),
  point: point(),
  path: path(),
  pgLsn: pgLsn(),
  polygon: polygon(),
  real: real(),
  float4: float4(),
  smallint: smallint(),
  int2: int2(),
  serial: serial(),
  serial4: serial4(),
  smallserial: smallserial(),
  serial2: serial2(),
  text: text(),
  time: time(),
  timeWithoutTimeZone: timeWithoutTimeZone(),
  timeWithTimeZone: timeWithTimeZone(),
  timestamp: timestamp(),
  timestamptz: timestamptz(),
  timestampWithoutTimeZone: timestampWithoutTimeZone(),
  timestampWithTimeZone: timestampWithTimeZone(),
  tsquery: tsquery(),
  tsvector: tsvector(),
  txidSnapshot: txidSnapshot(),
  uuid: uuid(),
  xml: xml(),
  enumType: enumType('my_enum-type', ['A', 'B', 'C'] as const),
});

const db = defineDb({ foo }, () => Promise.resolve({ rows: [], affectedCount: 0 }));

// @dts-jest:group data-types
{
  // @dts-jest:snap should select all data types
  toSnap(
    db
      .select(
        db.foo.int8,
        db.foo.bigint,
        db.foo.bigserial,
        db.foo.serial8,
        db.foo.bit,
        db.foo.bitVarying,
        db.foo.varbit,
        db.foo.boolean,
        db.foo.bool,
        db.foo.box,
        db.foo.bytea,
        db.foo.character,
        db.foo.char,
        db.foo.characterVarying,
        db.foo.varchar,
        db.foo.circle,
        db.foo.cidr,
        db.foo.caseInsensitiveText,
        db.foo.citext,
        db.foo.date,
        db.foo.doublePrecision,
        db.foo.float8,
        db.foo.inet,
        db.foo.decimal,
        db.foo.integer,
        db.foo.int,
        db.foo.int4,
        db.foo.interval,
        db.foo.json,
        db.foo.jsonb,
        db.foo.line,
        db.foo.lseg,
        db.foo.macaddr,
        db.foo.macaddr8,
        db.foo.money,
        db.foo.numeric,
        db.foo.point,
        db.foo.path,
        db.foo.pgLsn,
        db.foo.polygon,
        db.foo.real,
        db.foo.float4,
        db.foo.smallint,
        db.foo.int2,
        db.foo.serial,
        db.foo.serial4,
        db.foo.smallserial,
        db.foo.serial2,
        db.foo.text,
        db.foo.time,
        db.foo.timeWithoutTimeZone,
        db.foo.timeWithTimeZone,
        db.foo.timestamp,
        db.foo.timestamptz,
        db.foo.timestampWithoutTimeZone,
        db.foo.timestampWithTimeZone,
        db.foo.tsquery,
        db.foo.tsvector,
        db.foo.txidSnapshot,
        db.foo.uuid,
        db.foo.xml,
        db.foo.enumType,
      )
      .from(db.foo),
  );
}
