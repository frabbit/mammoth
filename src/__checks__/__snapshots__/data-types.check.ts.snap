// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`data-types should select all data types (type) should match snapshot 1`] = `"{ int8: GetDataType<string, false>; bigint: GetDataType<string, false>; bigserial: GetDataType<string, false>; serial8: GetDataType<string, false>; bit: GetDataType<string, false>; bitVarying: GetDataType<string, false>; varbit: GetDataType<string, false>; boolean: GetDataType<boolean, false>; bool: GetDataType<boolean, false>; box: GetDataType<string, false>; bytea: GetDataType<Buffer, false>; character: GetDataType<string, false>; char: GetDataType<string, false>; characterVarying: GetDataType<string, false>; varchar: GetDataType<string, false>; circle: GetDataType<{ x: number; y: number; radius: number; }, false>; cidr: GetDataType<string, false>; caseInsensitiveText: GetDataType<string, false>; citext: GetDataType<string, false>; date: GetDataType<Date, false>; doublePrecision: GetDataType<number, false>; float8: GetDataType<number, false>; inet: GetDataType<string, false>; decimal: GetDataType<string, false>; integer: GetDataType<number, false>; int: GetDataType<number, false>; int4: GetDataType<number, false>; interval: GetDataType<unknown, false>; json: GetDataType<unknown, false>; jsonb: GetDataType<unknown, false>; line: GetDataType<string, false>; lseg: GetDataType<string, false>; macaddr: GetDataType<string, false>; macaddr8: GetDataType<string, false>; money: GetDataType<string, false>; numeric: GetDataType<string, false>; point: GetDataType<{ x: number; y: number; }, false>; path: GetDataType<string, false>; pgLsn: GetDataType<string, false>; polygon: GetDataType<string, false>; real: GetDataType<number, false>; float4: GetDataType<number, false>; smallint: GetDataType<number, false>; int2: GetDataType<number, false>; serial: GetDataType<number, false>; serial4: GetDataType<number, false>; smallserial: GetDataType<number, false>; serial2: GetDataType<number, false>; text: GetDataType<string, false>; time: GetDataType<string, false>; timeWithoutTimeZone: GetDataType<string, false>; timeWithTimeZone: GetDataType<string, false>; timestamp: GetDataType<Date, false>; timestamptz: GetDataType<Date, false>; timestampWithoutTimeZone: GetDataType<Date, false>; timestampWithTimeZone: GetDataType<Date, false>; tsquery: GetDataType<string, false>; tsvector: GetDataType<string, false>; txidSnapshot: GetDataType<string, false>; uuid: GetDataType<string, false>; xml: GetDataType<string, false>; enumType: GetDataType<\\"A\\" | \\"B\\" | \\"C\\", false>; }"`;
